### PR TITLE
Use CourseOffering table as a source of family names

### PR DESCRIPTION
**Pull request by bethanyaconnor**
Use the course offering table as a source of family names. Unfortunately we can't remove ScriptConstants::FAMILY_NAMES yet, so we still need to check that list for family names.

I waffled between checking the CourseOffering table directly to see if a string matches a family name and generating a list of family names and looking in that. I ended up implementing the second option in order to distinguish between family names used for unit group and family names used for scripts. This means that this PR should not change any current behavior. 

This change does add a couple more offerings to the family name dropdown, at least locally. That seems fine, especially as more will be added soon. 

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/models/script.rb [hunk 1]
```diff
@@ -301,6 +301,10 @@ def all_scripts
       all_scripts.freeze
     end
 
+    def family_names
+      (CourseVersion.course_offering_keys('Script') + ScriptConstants::FAMILY_NAMES).uniq.sort
```
Comment (Hamms) [inline dashboard/app/models/script.rb]: `uniq` and `sort` are both complex operations; if performance is important enough here to cache `CourseOffering.find_by_key(key)`, I suspect it's important enough that we'd want to cache this, too

File: dashboard/app/models/course_version.rb [hunk 2]
```diff
@@ -105,4 +105,14 @@ def contained_lessons
...
     content_root_type == 'UnitGroup' ? standards_course_path(content_root) : standards_script_path(content_root)
   end
+
+  def self.should_cache?
+    Script.should_cache?
+  end
+
+  def self.course_offering_keys(content_root_type)
+    Rails.cache.fetch("course_version/course_offering_keys/#{content_root_type}", force: !should_cache?) do
+      CourseVersion.where(content_root_type: content_root_type).map {|cv| cv.course_offering&.key}.compact.uniq.sort
```
Comment (Hamms) [inline dashboard/app/models/course_version.rb]:
```suggestion
      CourseVersion.includes(:course_offering).where(content_root_type: content_root_type).map {|cv| cv.course_offering&.key}.compact.uniq.sort
```

File: dashboard/app/models/script.rb [hunk 3]
```diff
@@ -509,7 +513,7 @@ def self.get_without_cache(id_or_name, with_associated_models: true)
   #
   # @param id_or_name [String|Integer] script id, script name, or script family name.
   def self.get_from_cache(id_or_name)
-    if ScriptConstants::FAMILY_NAMES.include?(id_or_name)
+    if CourseOffering.get_from_cache(id_or_name)
```
Comment (Hamms) [inline dashboard/app/models/script.rb]:
```suggestion
    if CourseOffering.get_from_cache(id_or_name).present?
```

**Review by Hamms — APPROVED** [review]
couple minor nits, but generally LGTM!

File: dashboard/app/models/script.rb (hunk 1 above)
Comment (bethanyaconnor) [reply dashboard/app/models/script.rb]: Honestly not sure if `find_by_key` needs to be cached, it's just not a thing that changes once it hits production. `course_offering_keys` is cached but `FAMILY_NAMES` is large enough that I think you're right that caching is worth doing.

**Review by davidsbailey — APPROVED** [review]
LGTM!

**Pull request merged**